pkg/packaging/target.go

```diff
@@ -0,0 +1,57 @@
+package packaging
+
+import "fmt"
+
+// Target is the platform being targetted by the build. As "platform"
+// has several axis, we use a stuct to convey them.
+type Target struct {
+	Init     InitFlavor
+	Package  PackageFlavor
+	Platform PlatformFlavor
+}
+
+type InitFlavor string
+
+const (
+	LaunchD InitFlavor = "launchd"
+	SystemD            = "systemd"
+	Init               = "init"
+)
+
+type PlatformFlavor string
+
+const (
+	Darwin  PlatformFlavor = "darwin"
+	Windows                = "windows"
+	Linux                  = "linux"
+)
+
+type PackageFlavor string
+
+const (
+	Pkg PackageFlavor = "pkg"
+	Tar               = "tar"
+	Deb               = "deb"
+	Rpm               = "rpm"
+	Msi               = "msi"
+)
+
+func (t *Target) String() string {
+	return fmt.Sprintf("%s-%s-%s", t.Platform, t.Init, t.Package)
+}
+
+func (t *Target) Extension() string {
+	switch t.Package {
```

Comment:
you can return `t.Package` here since they map 1:1. Consider a `type PackageFlavor int` with iota and a String() method instead.

Reply:
`Target` is already a type, with methods. Feels off to have `PackageFlavor` be one. I stuck this here as switch anticipating some future need about changing the extension based on something more complicated than just a string(). But simplify for now,